epcis/src/main/java/io/openepcis/model/epcis/EPCISQueryDocument.java

```diff
@@ -82,6 +83,21 @@ public EPCISQueryDocument(EPCISQueryBody epcisBody) {
       this.context =
           getContextInfoFromEventList(epcisBody.getQueryResults().getResultsBody().getEventList());
     }
+
+    // Populating the namespaces directly from context during xml query
+    if (context != null && !context.isEmpty()) {
+      final DefaultJsonSchemaNamespaceURIResolver namespaceURIResolver =
+          DefaultJsonSchemaNamespaceURIResolver.getContext();
+      namespaceURIResolver.resetEventNamespaces();
+
+      for (Object item : context) {
+        if (item instanceof Map<?, ?>) {
+          final Map<String, String> namespaces = (Map<String, String>) item;
+          namespaces.forEach(
+              (key, value) -> namespaceURIResolver.populateEventNamespaces(value, key));
+        }
+      }
+    }
```

Comment:
duplicated same code from EPCISEventResponse
please try to come up with one single generic implemenation